src/sentry/snuba/metrics/metric_extraction.py

```diff
@@ -0,0 +1,258 @@
+import hashlib
+import logging
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Dict, Literal, Optional, Sequence, Tuple, TypedDict, Union, cast
+
+from snuba_sdk import BooleanCondition, Column, Condition
+from typing_extensions import NotRequired
+
+from sentry.constants import DataCategory
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.metrics.utils import MetricOperationType
+from sentry.snuba.models import SnubaQuery
+
+logger = logging.getLogger(__name__)
+
+# Name component of MRIs used for custom alert metrics.
+# TODO: Move to a common module.
+CUSTOM_ALERT_METRIC_NAME = "transactions/on_demand"
+QUERY_HASH_KEY = "query_hash"
+
+# Base type for conditions to evaluate on payloads.
+# TODO: Streamline with dynamic sampling.
+RuleCondition = Union["LogicalRuleCondition", "ComparingRuleCondition", "NotRuleCondition"]
+
+_SNUBA_TO_RELAY_FIELDS = {
+    "contexts[geo.country_code]": "event.geo.country_code",
+    "http_method": "event.http.method",
+    "release": "event.release",
+    "transaction_op": "event.transaction.op",
+    "transaction_status": "event.transaction.status",
+    "duration": "event.duration",
+    "measurements[cls]": "event.measurements.cls",
+    "measurements[fcp]": "event.measurements.fcp",
+    "measurements[fid]": "event.measurements.fid",
+    "measurements[fp]": "event.measurements.fp",
+    "measurements[lcp]": "event.measurements.lcp",
+    "measurements[ttfb]": "event.measurements.ttfb",
+    "measurements[ttfb.requesttime]": "event.measurements.ttfb.requesttime",
+    # TODO(ogi): Support fields whose resolution returns a function
+    # "browser.name": "event.browser.name",
+    # "http.status_code": "event.http.status_code",
+    # "os.name": "event.os.name",
+}
+
+_SNUBA_TO_METRIC_AGGREGATES: Dict[str, Optional[MetricOperationType]] = {
+    "count": "sum",
```

Comment:
I am a bit confused by this, is it because we somehow use the `sum` on a field which is always 1? I can't explain myself any other case in which they would be equivalent.

Reply:
The right side are aggregate operations on metrics, not on fields. `sum` therefore refers to aggregating a counter metric. 

In discover data model, `count` is always without parameters and counts the numbers of events. Translating this to metrics, there is a field less counter to do that job.